From bce47db9ba6e96914b6c5990779c4ad3727bbf90 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 20 Sep 2016 21:25:45 -0400
Subject: [PATCH 1029/1936] Add Workaround For Querying Voice Enable

When querying the voice enable on a kit which is disabled
you'd previously get unknown address warnings.
A special case has been added to show that the voices are
disabled.
---
 src/Misc/Master.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index abe21646..6df24ea7 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -731,6 +731,17 @@ bool Master::AudioOut(float *outr, float *outl)
         }
         ports.dispatch(msg, d, true);
         events++;
+        if(!d.matches) {
+            //workaround for requesting voice status
+            //gtk_hscale_new_with_range
+            //
+            int a=0, b=0, c=0;
+            char e=0;
+            if(4 == sscanf(msg, "/part%d/kit%d/adpars/VoicePar%d/Enable%c", &a, &b, &c, &e)) {
+                d.reply(msg, "F");
+                d.matches++;
+            }
+        }
         if(!d.matches) {// && !ports.apropos(msg)) {
             fprintf(stderr, "%c[%d;%d;%dm", 0x1B, 1, 7 + 30, 0 + 40);
             fprintf(stderr, "Unknown address<BACKEND> '%s:%s'\n", uToB->peak(), rtosc_argument_string(uToB->peak()));
-- 
2.47.0

